framework_tvb/tvb/interfaces/web/controllers/simulator/simulator_controller.py

```diff
@@ -490,6 +490,7 @@ def set_monitor_params(self, current_monitor_name, **data):
             previous_form_action_url=previous_form_action_url)
 
         form_action_url, if_bold_url = self.get_urls_for_next_monitor_fragment(next_monitor, current_monitor)
+        self.monitors_handler.all_monitors_dict[current_monitor.ui_name] = current_monitor
```

Comment:
the monitors_handler could have a method add_monitor(current_monitor) to fill its all_monitors_dict dictionary internally